### PR TITLE
chore(vclusterctl): check for pre 0.20 values

### PR DESCRIPTION
**Pull request by johannesfrey**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
Resolves ENG-3573

**Please provide a short message that should be published in the vcluster release notes**
Provide checks that abort "vcluster create" when using pre-v0.20 formatting in values.yaml.

**What else do we need to know?**

### CODE REVIEW

File: pkg/cli/create_helm.go [hunk 1]
```diff
@@ -148,17 +150,40 @@ func CreateHelm(ctx context.Context, options *CreateOptions, globalFlags *flags.
...
+				return err
+			}
+		} else {
+			currentValues, err = helmValuesYAML(release)
+			if err != nil {
+				return err
+			}
 		}
+
+		command := fmt.Sprintf("vcluster convert config --distro %s - <<EOF\n%s\nEOF", currentDistro, string(currentValues))
```
Comment (FabianKramm) [inline pkg/cli/create_helm.go:182]: This will print a big chunk, can we not do something like:
```
helm get values ... | vcluster convert config ...
```

File: pkg/cli/create_helm.go [hunk 2]
```diff
@@ -247,30 +278,6 @@ func CreateHelm(ctx context.Context, options *CreateOptions, globalFlags *flags.
 		return err
 	}
 
-	// check if vcluster already exists
-	if !cmd.Upgrade {
```
Comment (FabianKramm) [inline pkg/cli/create_helm.go:251]: Why is that removed?

**Review by FabianKramm — CHANGES_REQUESTED** [review]
See my comments

File: pkg/cli/create_helm.go (hunk 2 above)
Comment (johannesfrey) [reply pkg/cli/create_helm.go:251]: It looked like a duplicate of line 133-151.

File: pkg/cli/create_helm.go [hunk 3]
```diff
@@ -215,17 +240,23 @@ func CreateHelm(ctx context.Context, options *CreateOptions, globalFlags *flags.
...
 			}
-			return err
+			// TODO Delete after vCluster 0.19.x resp. the old config format is out of support.
+			// We cannot discriminate between k0s/k3s and eks/k8s. So we cannot prompt the actual values to convert, as this would cause false positives,
+			// because users are free to e.g. pass a k0s values file to a currently running k3s virtual cluster.
+			if isLegacyConfig(data) {
+				return fmt.Errorf("it appears you are using a vCluster configuration using pre-v0.20 formatting. Please run %q to convert the values to the latest format", "vcluster convert config")
+			}
+			// TODO end
 		}
```
Comment (facchettos) [inline pkg/cli/create_helm.go:253]: I would maybe structure the if differently here 
```go
if errors.Is(err, config.ErrInvalidConfig) {
  if isLegacyConfig(data){
    return fmt.Errorf("it appears you are using a vCluster configuration using pre-v0.20 formatting. Please run %q to convert the values to the latest format", "vcluster convert config")
  }
} else if err != nil {
  return err
}
```
that is simpler to follow imho

File: pkg/cli/create_helm.go (hunk 1 above)
Comment (johannesfrey) [reply pkg/cli/create_helm.go:182]: Sure. Just to be sure. `helm get values` will retrieve the latest revision right? I tested locally and it looks like it 🤔

File: pkg/cli/create_helm.go [hunk 4]
```diff
@@ -320,35 +327,66 @@ func isVClusterDeployed(release *helm.Release) bool {
...
+	if err := yaml.Unmarshal(releaseRaw, &helmChart); err != nil {
+		return nil, err
+	}
+
+	values, err := yaml.Marshal(helmChart.Chart.Values)
+	if err != nil {
+		return nil, err
+	}
+
+	return values, nil
```
Comment (facchettos) [inline pkg/cli/create_helm.go:388]: here you can just do 
```go
return yaml.Marshal(helmChart.Chart.Values)
```

File: pkg/cli/create_helm.go [hunk 5]
```diff
@@ -320,35 +327,66 @@ func isVClusterDeployed(release *helm.Release) bool {
...
+	helmChart := struct {
+		Chart struct {
+			Values map[string]interface{} `json:"values"`
+		} `json:"chart"`
+	}{}
 
-// TODO end
+	if err := yaml.Unmarshal(releaseRaw, &helmChart); err != nil {
+		return nil, err
+	}
```
Comment (facchettos) [inline pkg/cli/create_helm.go:381]: can we use an encoder/decoder here instead of reading everything into memory or not?

File: pkg/cli/create_helm.go (hunk 4 above)
Comment (johannesfrey) [reply pkg/cli/create_helm.go:388]: When going with [Fabian's suggestion](https://github.com/loft-sh/vcluster/pull/1738#discussion_r1592351094) this func will be removed.

File: pkg/cli/create_helm.go [hunk 6]
```diff
@@ -148,17 +147,31 @@ func CreateHelm(ctx context.Context, options *CreateOptions, globalFlags *flags.
...
-	// TODO Delete after vCluster 0.19.x resp. the old config format is out of support.
-	if isVClusterDeployed(release) {
-		migratedValues, err := migrateLegacyHelmValues(release)
-		if err != nil {
-			return err
+	// TODO Refactor after vCluster 0.19.x resp. the old config format is out of support.
+	// Early abort if a user runs a virtual cluster < v0.20 without providing any values files during an upgrade.
+	// We do this because we don't want to "automagically" convert the old config implicitly, without the user
+	// realizing that the virtual cluster is running with the old config format.
+	if isVClusterDeployed(release) && isLegacyVCluster(release.Chart.Metadata.Version) && len(cmd.Values) <= 0 {
```
Comment (facchettos) [inline pkg/cli/create_helm.go:154]: little nit but I don't believe Len can return something less than 0

File: pkg/cli/create_helm.go (hunk 3 above)
Comment (johannesfrey) [reply pkg/cli/create_helm.go]: **Update**: The `ErrInvalidConfig` did not provide any benefit. It was actually a left-over from when the config unmarshaling looked different. So I changed it to a simple err check.

**Review by FabianKramm — APPROVED** [review]
